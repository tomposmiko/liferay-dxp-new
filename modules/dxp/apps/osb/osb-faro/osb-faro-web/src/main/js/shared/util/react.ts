import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
