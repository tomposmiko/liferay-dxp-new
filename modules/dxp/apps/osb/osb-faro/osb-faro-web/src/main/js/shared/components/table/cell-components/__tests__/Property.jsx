import PropertyCell from '../Property';
import React from 'react';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

describe('PropertyCell', () => {
	it('should render', () => {
		const {container} = render(
			<PropertyCell
				data={{
					name: 'email',
					value: 'TestTest@liferay.com'
				}}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('should render - when has no value', () => {
		const {getByText} = render(
			<PropertyCell
				data={{
					name: 'email',
					value: ''
				}}
			/>
		);

		expect(getByText('-')).toBeTruthy();
	});
});
