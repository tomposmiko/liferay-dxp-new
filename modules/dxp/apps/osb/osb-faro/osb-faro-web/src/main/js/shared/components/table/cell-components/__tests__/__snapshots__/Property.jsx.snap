// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCell should render 1`] = `
<div>
  <td
    class="property-cell"
  >
    <div
      class="name"
    >
      email
    </div>
    <div
      class="table-title"
    >
      TestTest@liferay.com
    </div>
  </td>
</div>
`;
